src/App.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { ChangeEvent, KeyboardEvent } from "react";
+
+interface CounterProps {
+  initialValue: number;
+}
+
+interface CounterState {
+  value: number;
+}
+
+class Counter extends React.Component<CounterProps> {
+  state: CounterState;
+
+  constructor(props: CounterProps) {
+    super(props);
+    this.state = {
+      value: props.initialValue,
+    };
+  }
+
+  increment = () => {
+    this.setState({ value: this.state.value + 1 });
+  };
+
+  decrement = () => {
+    this.setState({ value: this.state.value - 1 });
+  };
+
+  render() {
+    return React.createElement(
+      "div",
+      { id: "counter" },
+      React.createElement("button", { onClick: this.decrement }, `-`),
+      React.createElement("span", null, `${this.state.value}`),
+      React.createElement("button", { onClick: this.increment }, `+`)
+    );
+  }
+}
+
+interface SearchFormProps {
+  initialSearchQuery: string;
+  onSearch: (query: string) => void;
+}
+
+interface SearchFormState {
+  searchQuery: string;
+}
+
+class SearchForm extends React.Component<SearchFormProps> {
+  state: SearchFormState;
+
+  constructor(props: SearchFormProps) {
+    super(props);
+    this.state = {
+      searchQuery: props.initialSearchQuery,
+    };
+  }
+
+  search = () => {
+    this.props.onSearch(this.state.searchQuery);
+  };
+
+  change = (event: ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      searchQuery: event.target.value,
+    });
+  };
+
+  keyDown = (event: KeyboardEvent<HTMLElement>) => {
```

Comment:
Just a note, no action needed: if you use `form` instead of `div` on line 77 `<div id="searchForm">` then pressing Enter key will automatically run `onSubmit` of the form.
On the flip side `<button type="submit" />` should be nested inside a form, clicking such button will automatically submit its parent form. If you need an arbitrary button not associated with a form use `<button type="button" />`.

Reply:
I made the update, and now it's only using one submit event instead of click and enter. Thanks for the tip.